lua/yazi.lua

```diff
@@ -37,13 +37,20 @@ local function on_exit(job_id, code, event)
 	vim.cmd("silent! :checktime")
 
 	if vim.api.nvim_win_is_valid(prev_win) then
-		vim.api.nvim_win_close(win, true)
+		if vim.api.nvim_win_is_valid(win) then
+			vim.api.nvim_win_close(win, true)
+		end
 		vim.api.nvim_set_current_win(prev_win)
 		if code == 0 and file_exists(output_path) == true then
 			local chosen_file = vim.fn.readfile(output_path)[1]
 			if chosen_file then
 				vim.cmd(string.format('edit %s', chosen_file))
 			end
+
+			-- if open a folder, auto switch nvim working directory
+			if vim.fn.isdirectory(chosen_file) == 1 then
+				vim.cmd('cd ' .. chosen_file)
+			end
```

Comment:
Thank You :) for your reply, I noticed the main change, etc.. and that pull-request doesn't make sense anymore.. this was created from the master branch .. so I think this could cause issues ..

I will be creating a different PULL-REQUEST based on the new `main` branch